### PR TITLE
Use kl_coeff in loss function calculation

### LINKED ISSUE #328: On the VAE model, `kl_coeff` is never applied in the loss function
## 🐛 Bug
The VAE model contained here takes an optional `kl_coeff` parameter that's supposed to be a scaling factor for the KL term of the variational loss. This might be useful to avoid the ["posterior collapse phenomenon"](https://datascience.stackexchange.com/questions/48962/what-is-posterior-collapse-phenomenon). However, while inspecting the codebase (i.e. [here](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py)) I can see that the parameter is initialized, stored, but never used.

### To Reproduce

1. Train a VAE
2. Set a custom value for the `kl_coeff` argument
3. Train again

#### Code sample

Code for achieving this can be taken from https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/tests/models/test_autoencoders.py#L13 but the behavior should be obvious from `basic_vae_module.py`

### Expected behavior

The loss function calculation (https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/pl_bolts/models/autoencoders/basic_vae/basic_vae_module.py#L119) should account for `kl_loss`.

### Environment

 - PyTorch Version (e.g., 1.0): 1.8 (nightly)
 - OS (e.g., Linux): Linux
 - How you installed PyTorch (`conda`, `pip`, source): conda
 - Build command you used (if compiling from source): N/A
 - Python version: 3.8
 - CUDA/cuDNN version: N/A
 - GPU models and configuration: N/A
 - Any other relevant information: N/A

### Additional context

Thanks for the amazing library/libraries! :)

### PR DESCRIPTION
**Pull request by miccio-dk**
## What does this PR do?

The VAE model contained here takes an optional `kl_coeff` parameter that's supposed to be a scaling factor for the KL term of the variational loss. This might be useful to avoid the "posterior collapse phenomenon". This pull request makes sure that it's used in the loss function.

Fixes #328.

## Before submitting
- [x] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together? Otherwise, we ask you to create a separate PR for every change.
- [ ] Did you make sure to update the documentation with your changes? _not needed_
- [ ] Did you write any new necessary tests? _not needed_
- [ ] Did you verify new and existing tests pass locally with your changes? _no changes in tests_
- [ ] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning-bolts/blob/master/CHANGELOG.md)? _it's a bug fix, no new feature_

## PR review
 - [x] Is this pull request ready for review? (if not, please submit in draft mode)

Anyone in the community is free to review the PR once the tests have passed.
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Absolutely! =)

### CODE REVIEW

**Review by akihironitta — APPROVED** [review]
LGTM!

**Review by ananyahjha93 — APPROVED** [review]

**Comment by miccio-dk** [comment]
My pleasure! Excuse the weird question, but would you be able to give me an estimate of when this will be included in a minor release? Thanks! :)

**Comment by ananyahjha93** [comment]
we'll have a release as soon as we are done with simclr fixes, which should be a couple of days max.

**Review by teddykoker — APPROVED** [review]
LGTM
